client/src/components/saveModal.jsx

```diff
@@ -0,0 +1,310 @@
+  margin-bottom: 13px;
+  border: 1px solid rgb(153, 153, 153);
+  border-radius: 3px;
+  padding: 5px;
+  box-sizing: border-box;  
+`
+const ZipCodeInput = styled.input`
+  height: 30px;
+  width: 100%;
+  margin-bottom: 10px;
+  border: 1px solid rgb(153, 153, 153);
+  border-radius: 3px;
+  padding: 5px;
+  box-sizing: border-box;  
+`
+
+const FirstNameInput = styled.input`
+  height: 18px;
+  width: 50%;
+  display: inline;
+  border: 1px solid rgb(153, 153, 153);
+  border-radius: 3px;
+  padding: 5px;
+  margin-right: 10px;
+`
+const LastNameInput = styled.input`
+  height: 18px;
+  width: 50%;
+  display: inline;
+  border: 1px solid rgb(153, 153, 153);
+  border-radius: 3px;
+  padding: 5px;
+`
+
+const SubTitle = styled.p`
+  display: inline; 
+  color: #999999; 
+  font-size: .75rem;
+  font-weight: normal;
+  text-align: right;
+`
+const ShareModalButton = styled.div`
+  width: 100%;
+  box-sizing: border-box;
+  text-align: center; 
+  border-radius: 3px;
+  box-shadow: rgba(0, 0, 0, 0.3) 0px 1px 1px 0px;
+  border-style: solid 1px;
+  padding: 8px;
+  margin-right: 12px;
+  margin-bottom: 12px;
+  background-color: #DF1201;
+  background: linear-gradient(#d90007, #c91400);
+  color: white;
+  font-size: 14px;
+  font-weight: bold;
+  border: solid 1px;
+  border-color: #8d0005;
+  &:hover{
+    cursor: pointer;
+    background: #ff1a1a;
+  }
+`
+const SubHeading = styled.p`
+  color: black;
+  font-weight: bold;
+  margin-bottom: 6px;
+`
+
+const TOSandPrivacyStatement = styled.p`
+  color: #333;
+  margin-bottom: 6px;
+  font-size: 12px;
+  text-align: center;
+`
+const ModalBody = styled.div`
+  margin: 10px 30px 20px;
+`
+
+const TOSandPrivacyLink = styled.a`
+  color: #0073BB;
+  text-decoration: none;
+  &:hover{
+    text-decoration: underline;
+  }
+`
+const NameContainer = styled.div`
+  display: flex;
+  margin-bottom: 6px;
+`
+const BirthdayContainer = styled.div`
+  display: flex;
+  margin-bottom: 15px;
+`
+const BirthdayMonthSelect = styled.select`
+  margin-right: 5px;  
+  width: 33.33%;
+  height: 100%;
+  display: inline;
+  border: 1px solid rgb(153, 153, 153);
+  border-radius: 3px;
+  padding: 5px;
+`
+const BirthdayDaySelect = styled.select`
+  width: 33.33%;
+  height: 100%;
+  display: inline;
+  border: 1px solid rgb(153, 153, 153);
+  border-radius: 3px;
+  padding: 5px;
+`
+const BirthdayYearSelect = styled.select`
+  margin-left: 5px;
+  width: 33.33%;
+  height: 100%;
+  display: inline;
+  border: 1px solid rgb(153, 153, 153);
+  border-radius: 3px;
+  padding: 5px;
+`
+
+const OnYelp = styled.div`
+  text-align: right;
+`
+
+const SaveModal = (props) => {
+  
+  let months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug','Sep', 'Nov', 'Dec'];
+  let renderMonths = [];
+  for (let i = 0; i < months.length; i++){
+    renderMonths.push(<option value={months[i]}>{months[i]}</option>)
+  }
+
+  let days = [];
+  for (let i = 1; i < 32; i++){
+    days.push(<option value={i}>{i}</option>)
+  }
```

Comment:
thanks for making this generate rather than coding it all :)

Reply:
nah thanks for reminding me, it is much, much cleaner and way more maintainable this way!